tests/production/DeleteUserSettingsFromCouchTests.js

```diff
@@ -0,0 +1,220 @@
+            listener: "gpii.tests.productionConfigTesting.testGetThenSaveDocForDeletion"
+        }, {
+            func: "gpii.tests.productionConfigTesting.getPrefsSafeDoc",
+            args: ["{getSettingsUserPrefsSafe}", "{getSettingsUserKey}" ]
+        }, {
+            event: "{getSettingsUserPrefsSafe}.events.onComplete",
+            listener: "gpii.tests.productionConfigTesting.testGetThenSaveDocForDeletion"
+        }, {
+            func: "{getGpiiKeyNoPrefsSafe}.send",
+            args: [null, { port: "5984" }]
+        }, {
+            event: "{getGpiiKeyNoPrefsSafe}.events.onComplete",
+            listener: "gpii.tests.productionConfigTesting.testGetThenSaveDocForDeletion"
+        }, {
+            funcName: "gpii.tests.productionConfigTesting.getPrefsSafeDoc",
+            args: ["{getGpiiKeyNoPrefsSafePrefsSafe}", "{getGpiiKeyNoPrefsSafe}"]
+        }, {
+            event: "{getGpiiKeyNoPrefsSafePrefsSafe}.events.onComplete",
+            listener: "gpii.tests.productionConfigTesting.testGetThenSaveDocForDeletion"
+        }, {
+            func: "{getNonExistentGpiiKey}.send",
+            args: [null, { port: "5984" }]
+        }, {
+            event: "{getNonExistentGpiiKey}.events.onComplete",
+            listener: "gpii.tests.productionConfigTesting.testGetThenSaveDocForDeletion"
+        }, {
+            funcName: "gpii.tests.productionConfigTesting.getPrefsSafeDoc",
+            args: ["{getNonExistentGpiiKeyPrefsSafe}", "{getNonExistentGpiiKey}"]
+        }, {
+            event: "{getNonExistentGpiiKeyPrefsSafe}.events.onComplete",
+            listener: "gpii.tests.productionConfigTesting.testGetThenSaveDocForDeletion"
+        }, {
+            funcName: "gpii.tests.productionConfigTesting.bulkDelete",
+            args: [
+                "{deleteInBulk}",
+                [
+                    "{getSettingsUserKey}.options.docToRemove",
+                    "{getSettingsUserPrefsSafe}.options.docToRemove",
+                    "{getGpiiKeyNoPrefsSafe}.options.docToRemove",
+                    "{getGpiiKeyNoPrefsSafePrefsSafe}.options.docToRemove",
+                    "{getNonExistentGpiiKey}.options.docToRemove",
+                    "{getNonExistentGpiiKeyPrefsSafe}.options.docToRemove"
+                ]
+            ]
+        }, {
+            event: "{deleteInBulk}.events.onComplete",
+            listener: "gpii.tests.productionConfigTesting.testStatusCode"
+        },
+        { funcName: "fluid.log", args: ["Deleted 'user' test prefs safes"]}
+    ]
+});
+
+fluid.defaults("gpii.tests.productionConfigTesting.deleteRecordsSequence", {
+    gradeNames: ["fluid.test.sequence"],
+    sequenceElements: {
+        deleteRecords: {
+            gradeNames: "gpii.tests.productionConfigTesting.deleteUserSettings"
+        }
+    }
+});
+
+// Tests for deleting test 'user' preferences from the database
+gpii.tests.productionConfigTesting.deleteTestRecordsFromDatabaseTests = [{
```

Comment:
Sorry I should have thought about this much earlier rather than discovering it now. Long story in short, only deleting these records is not enough.

If you look into the CouchDB after running "npm run test:vagarantProduction", there are still some new records in there:
<img width="683" alt="Screen Shot 2019-03-29 at 2 36 05 PM" src="https://user-images.githubusercontent.com/572877/55255108-2427b280-5230-11e9-8ba4-f96257341758.png">

Open one of them, it looks like below and all these new records are in this structure:
<img width="605" alt="Screen Shot 2019-03-29 at 2 36 46 PM" src="https://user-images.githubusercontent.com/572877/55255113-25f17600-5230-11e9-8eee-b7845319bfb0.png">

These are access tokens assigned to OAuth clients. Refer to [the documentation](https://wiki.gpii.net/w/GPII_OAuth_2_Guide#Resource_Owner_GPII_Key_Grant) about OAuth authentication GPII supports. Every GET or PUT requests to /settings endpoint requires an valid access token first. This is when they are created. They need to be removed from the database too when the production test completes.

Reply:
Followup, based on a teleconference with @cindyli 

- Add a new view to `_design/views` that will find these access tokens based on:
-- `gpiiKey`
-- `clientCredentials`
-- `type` (always "`gpiiAppInstallationAuthorization`")

(Note: The `gpiiKey` and `clientCredentials` are the same as those used in the test scenario that created the access_token in the first place.)

Also, this new view will be of use for [GPII-3646](https://issues.gpii.net/browse/GPII-3646) where we free up the database of all the expired access tokens created from day-to-day use of the system.